### DISCUSSION TITLE
How to create a structure element, When I use the erode() and dilate()?

### DISCUSSION THREAD

**Discussion by Lenlend** [libvips/libvips#3388] [Q&A] [answered]
I want to use the interface of morph by c++, but I can't find the interface to create a mask. Is that existed  a function to create the structure element, just like the cv::getStructureElement() in OpenCV?

**Comment by jcupitt** [comment, marked as answer]
Hi @Lenlend,

libvips uses images as masks, so you create them like you'd create any image. There are some convenience functions to make them in code, see:

https://www.libvips.org/API/8.14/cpp/classVImage.html#ac4166259736c14dbd799b7ca98c5a72d

Eg.:

```C++
VImage mask = VImage::new_matrixv(3, 3,
    128.0, 255.0, 128.0,
    255.0, 255.0, 255.0,
    128.0, 255.0, 128.0);
```

We ought to add something more C++11-like for this, but no one's got around to it yet.
